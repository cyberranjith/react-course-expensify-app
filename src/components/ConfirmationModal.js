import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => (
    <Modal
        isOpen={props.showModal}
        onRequestClose={props.closeConfirmation}
        contentLabel={props.confirmationText}
        closeTimeoutMS={200}
        ariaHideApp={false}
        className="modal"
    >
        <h3 className="modal__title">{props.confirmationText}</h3>
        <button onClick={props.performConfirmedAction} className="modal__button">Yes</button>
        <button onClick={props.closeConfirmation} className="modal__button">No</button>
    </Modal>
);

export default ConfirmationModal;