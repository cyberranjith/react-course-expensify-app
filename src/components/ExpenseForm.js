import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            description: props.expense ? props.expense.description : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            note: props.expense ? props.expense.note : '',
            error: ''
        };
    };
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState( () => ( { description: description } ) );
    };
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState( () => ( { note: note } ) );
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState( () => ( { amount: amount } ) );
        }
    };
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ( {createdAt: createdAt} ));
        }        
    };
    onFocusChange = ( {focused} ) => {
        this.setState(() => ({calendarFocused: focused}) );
    };
    onFormSubmit = (e) => {
        e.preventDefault();

        if (!this.state.description || !this.state.amount) {
            this.setState(() => ({error: 'Please provide Description and Amount!'}));
        } else {
            this.setState(() => ({error: ''}));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            });
        }
    };
    render() {
        return (              
            <form className="form" onSubmit={this.onFormSubmit}>
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                <input
                    type="text"
                    name="description"
                    placeholder="Description"
                    className="text-input"
                    value={this.state.description}
                    onChange={this.onDescriptionChange}
                    autoFocus   
                />
                <input
                    type="text"
                    name="amount"
                    placeholder="Amount"
                    className="text-input"
                    value={this.state.amount}
                    onChange={this.onAmountChange}
                />
                <SingleDatePicker 
                    date = {this.state.createdAt}
                    onDateChange = {this.onDateChange}
                    focused = {this.state.calendarFocused}
                    onFocusChange = {this.onFocusChange}                        
                    numberOfMonths={1}
                    isOutsideRange={()=> false}
                />
                <textarea
                    type="textarea"
                    name="note"
                    placeholder="Add a note about your expense"
                    className="text-area"
                    value={this.state.note}
                    onChange={this.onNoteChange}
                />
                <div>
                    <button className="button">Save Expense</button>
                </div>                
            </form>
        )
    }
};