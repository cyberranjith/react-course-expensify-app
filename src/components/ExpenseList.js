import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';

import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expense Found</p>
            ) : (
                props.expenses.map((expense) => {
                    return (
                        <ExpenseListItem key={expense.id} expense={expense} />
                    );
                })
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

// Return a HOC which is connected to the redux state
export default connect(mapStateToProps)(ExpenseList);