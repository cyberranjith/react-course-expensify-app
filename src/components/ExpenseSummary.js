import React from 'react';
import {connect} from 'react-redux';

import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ expenseCount, expenseTotal }) => {
    const formattedExpenseTotal = numeral(expenseTotal / 100).format('$0,0.00');
    return (
        <div>
            { 
                expenseCount === 0 ? 
                ( <h1> No expense returned </h1> ) :
                ( <h1> Viewing {expenseCount} expense(s) totalling {formattedExpenseTotal} </h1> )
            }
        </div>
    );
    
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: getExpensesTotal(visibleExpenses)
    }
};

// Return a HOC which is connected to the redux state
export default connect(mapStateToProps)(ExpenseSummary);