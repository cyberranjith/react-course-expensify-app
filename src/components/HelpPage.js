import React from 'react';

const HelpPage = () => (
    <div>
        From my help component!
    </div>
);

export default HelpPage;