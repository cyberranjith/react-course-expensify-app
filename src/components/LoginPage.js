import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin, startGitLogin } from '../actions/auth';

export const LoginPage = ( {startGoogleLogin, startGitLogin} ) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className="button" onClick={startGoogleLogin}>Login with Google</button>
            <button className="button" onClick={startGitLogin}>Login with Git</button>
        </div>        
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startGitLogin: () => dispatch(startGitLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);