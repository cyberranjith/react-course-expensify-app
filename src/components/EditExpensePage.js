import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses'
import { removeExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onFormSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    onRemoveExpenseButtonClick = (e) => {
        this.props.removeExpense(this.props.expense);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onFormSubmit}
                />
                <button onClick={this.onRemoveExpenseButtonClick}>
                    Remove
                </button>
            </div>
            
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find( (expense) => expense.id === props.match.params.id )
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        editExpense: (id, expense) => {
            dispatch(editExpense(id, expense));
        },
        removeExpense: (expense) => {
            dispatch(removeExpense(expense));
        } 
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);