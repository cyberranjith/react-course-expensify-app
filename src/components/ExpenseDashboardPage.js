import React from 'react';
import ExepenseList from './ExpenseList';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;