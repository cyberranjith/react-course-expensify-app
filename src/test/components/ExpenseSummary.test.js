import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary';

test('should render Expense Summary with three expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expenseTotal={11456}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render Expense Summary with one expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={1000}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render Expense Summary with no expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={0} expenseTotal={0}/>);
    expect(wrapper).toMatchSnapshot();
});
