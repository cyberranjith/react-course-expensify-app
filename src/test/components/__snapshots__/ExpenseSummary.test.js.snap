// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense Summary with no expense 1`] = `
<div>
  <p>
     No expense returned 
  </p>
</div>
`;

exports[`should render Expense Summary with one expense 1`] = `
<div>
  <p>
     Viewing 
    1
     expense(s) totalling 
    $10.00
     
  </p>
</div>
`;

exports[`should render Expense Summary with three expense 1`] = `
<div>
  <p>
     Viewing 
    3
     expense(s) totalling 
    $114.56
     
  </p>
</div>
`;
